Sources/SwiftIfConfig/IfConfigEvaluation.swift

```diff
@@ -627,6 +630,57 @@ private func diagnoseLikelySimulatorEnvironmentTest(
   return IfConfigDiagnostic.likelySimulatorPlatform(syntax: ExprSyntax(binOp)).asDiagnostic
 }
 
+/// If this identifier looks like it is a `TARGET_OS_*` compilation condition,
+/// produce a diagnostic that suggests replacing it with the `os(*)` syntax.
+///
+/// For example, this checks for conditions like:
+///
+/// ```
+/// #if TARGET_OS_IOS
+/// ```
+///
+/// which should be replaced with
+///
+/// ```
+/// #if os(iOS)
+/// ```
+private func diagnoseLikelyTargetOSTest(
+  at reference: DeclReferenceExprSyntax,
+  name: String
+) -> Diagnostic? {
+  let prefix = "TARGET_OS_"
+  guard name.hasPrefix(prefix) else { return nil }
+  let osName = String(name.dropFirst(prefix.count))
+
+  if unmappedTargetOSNames.contains(osName) {
+    return IfConfigDiagnostic.likelyTargetOS(syntax: ExprSyntax(reference), replacement: nil).asDiagnostic
+  }
+
+  guard let (function, argument) = targetOSNameMap[osName] else { return nil }
+  let replacement = FunctionCallExprSyntax(callee: DeclReferenceExprSyntax(baseName: .identifier(function))) {
+    LabeledExprSyntax(expression: DeclReferenceExprSyntax(baseName: .identifier(argument)))
+  }
+
+  return IfConfigDiagnostic.likelyTargetOS(
+    syntax: ExprSyntax(reference),
+    replacement: ExprSyntax(replacement)
+  ).asDiagnostic
+}
+
+// TARGET_OS_* macros that don’t have a direct Swift equivalent
+private let unmappedTargetOSNames = ["WIN32", "UNIX", "MAC", "IPHONE", "EMBEDDED"]
+private let targetOSNameMap: [String: (function: String, argument: String)] = [
+  "WINDOWS": ("os", "Windows"),
+  "LINUX": ("os", "Linux"),
+  "OSX": ("os", "macOS"),
+  "IOS": ("os", "iOS"),
+  "MACCATALYST": ("targetEnvironment", "macCatalyst"),
+  "TV": ("os", "tvOS"),
+  "WATCH": ("os", "watchOS"),
+  "VISION": ("os", "visionOS"),
+  "SIMULATOR": ("targetEnvironment", "simulator"),
+]
```

Comment:
Would it be simpler if the value of the dictionary is an `ExprSyntax` directly and then construct that from a string literal?

```suggestion
private let targetOSNameMap: [String: ExprSyntax] = [
  "WINDOWS": "os(Windows)",
  ...
]
```